### PR TITLE
fix(npm): don't delete lockfiles early

### PR DESCRIPTION
**Pull request by viceice**
## Changes
Don't delete lockfiles early, otherwise we can't use them to extract contraints.
They will be deleted before running the manager anyways (which was a no-op before this)

compare with ignore whitespace

## Context
- #28470
- https://github.com/renovate-reproductions/28467-pnpm-v9 (only tested for pnpm)

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/npm/post-update/index.ts [hunk 1]
```diff
@@ -141,10 +141,7 @@ export async function writeExistingFiles(
     const npmLock = packageFile.managerData.npmLock;
     if (npmLock) {
       const npmLockPath = npmLock;
-      if (
-        process.env.RENOVATE_REUSE_PACKAGE_LOCK === 'false' ||
-        config.reuseLockFiles === false
-      ) {
+      if (process.env.RENOVATE_REUSE_PACKAGE_LOCK === 'false') {
```
Comment (viceice) [inline lib/modules/manager/npm/post-update/index.ts:144]: @rarkins i think we should remove this

Comment (viceice) [reply lib/modules/manager/npm/post-update/index.ts:144]: I'm not sure what the else path is doing. It's somehow manipulating the lockfile (widens some deps)

**Comment by viceice** [comment]
compare with ignore whitespace

File: lib/modules/manager/npm/extract/index.spec.ts
Comment (rarkins) [inline lib/modules/manager/npm/extract/index.spec.ts:1]: What difference do the changes in these files make?

**Review by rarkins — APPROVED** [review]

File: lib/modules/manager/npm/extract/index.spec.ts
Comment (viceice) [reply lib/modules/manager/npm/extract/index.spec.ts:1]: just test coverage for functions

**Pull request merged**